metabase/metabase
# Fix expressions with the same name as their base column

Fixes https://github.com/metabase/metabase/issues/40064

Removes a FE hack for detecting cycles and relies on MBQL lib instead. To make it work, pass `expressionPosition`.

How to verify:
- New -> Question -> Orders
- Custom expression -> `[Tax] * 2`, name `Tax`
- Click on the expression and change it to `[Tax] * 3` without renaming
- You should be able to save and visualize the question

<img width="578" alt="Screenshot 2024-06-24 at 16 07 44" src="https://github.com/metabase/metabase/assets/8542534/9d30dce8-d3d8-4e23-8c2e-79d757f6133c">
